### ISSUE TITLE
Khmer test cases

### ISSUE THREAD

**Issue by veer66** [veer66/chamkho#4]

**Comment by Vichet97** [comment]
Hi, So what do you use for cutting ? Or you create your own logic ?

**Comment by veer66** [comment]
> Hi, So what do you use for cutting ? Or you create your own logic ?

Chamkho used this library https://github.com/veer66/wordcut-engine, which is ported from http://thaiwordseg.sourceforge.net. 

wordcut-engine algorithm is based on 3 things.

1. word list
2. cluster rules
3. split rules

We have a Khmer word list at https://github.com/veer66/chamkho/blob/main/data/khmerdict.txt. However, cluster rules and split rules are now only in Thai. 

* https://github.com/veer66/wordcut-engine/blob/main/data/thai_split_rules.txt
* https://github.com/veer66/chamkho/blob/main/data/thai_cluster_rules.txt

Adding rules more Khmer shouldn't be difficult since they are ordinary regular expressions. I implemented most of these. However, they are not my logic. Word-list-based algorithm was proposed by Yuen Poovarawan in the early 80s. Using cluster rules was proposed in 2000 in SNLP at Chaingmai, but I forgot the author names. Using prohibit rules was Chanathip Thumkanon's idea 20 years ago.

**Comment by Vichet97** [comment]
I believe there's already library for this work

**Comment by veer66** [comment]
> I believe there's already library for this work

Yes, my library has been available since 2002. Compared to ICU, Chamkho is easier to customize. I only need people to verify tokenization results for languages besides Thai and Lao.

**Comment by Vichet97** [comment]
Your split cutting is already from what you just showed above
